jolt-core/src/jolt/vm/read_write_memory.rs

```diff
@@ -68,21 +69,35 @@ impl<F: PrimeField, G: CurveGroup<ScalarField = F>> ReadWriteMemory<F, G> {
     for memory_access in memory_trace {
       match memory_access {
         MemoryOp::Read(a, v) => {
-          a_read_write.push(a);
+          assert!(a < REGISTER_COUNT || a >= RAM_START_ADDRESS);
+          let remapped_a = if a >= RAM_START_ADDRESS {
+            a - RAM_START_ADDRESS + REGISTER_COUNT
+          } else {
+            a
+          };
+          debug_assert_eq!(v, v_final[remapped_a as usize]);
+          a_read_write.push(remapped_a);
           v_read.push(v);
           v_write.push(v);
-          t_read.push(t_final[a as usize]);
+          t_read.push(t_final[remapped_a as usize]);
           t_write.push(timestamp + 1);
-          t_final[a as usize] = timestamp + 1;
+          t_final[remapped_a as usize] = timestamp + 1;
         }
         MemoryOp::Write(a, v_old, v_new) => {
-          a_read_write.push(a);
+          assert!(a < REGISTER_COUNT || a >= RAM_START_ADDRESS);
+          let remapped_a = if a >= RAM_START_ADDRESS {
```

Comment:
What is the case where we don't remap? Registers?

If so could you add a comment to that effect?